custom_components/ocpp/api.py

```diff
@@ -464,30 +467,54 @@ async def handle_set_charge_rate(call):
             all_measurands = self.entry.data.get(
                 CONF_MONITORED_VARIABLES, DEFAULT_MEASURAND
             )
+            autodetect_measurands = self.entry.data.get(
+                CONF_MONITORED_VARIABLES_AUTOCONFIG,
+                DEFAULT_MONITORED_VARIABLES_AUTOCONFIG,
+            )
+            quirk_monitored_to_charger = self.entry.data.get(
+                CONF_QUIRK_MONITORED_TO_CHARGER,
+                DEFAULT_QUIRK_MONITORED_TO_CHARGER,
+            )
+
             key = ckey.meter_values_sampled_data.value
-            try:
-                chgr_measurands = await self.get_configuration(key)
-            except Exception:
-                _LOGGER.debug(
-                    f"'{self.id}' had error while returning measurands, ignoring"
+
+            # Quirk:
+            # Workaround for a bug on chargers that have invalid MeterValuesSampledData
+            # configuration. This quirk allows to set it to the user provided list.
+            # Corresponding issue: https://github.com/lbbrhzn/ocpp/issues/1275
+            if quirk_monitored_to_charger:
+                _LOGGER.info(
+                    f"'{self.id}' quirk: set OCPP Measurands on charger to '{all_measurands}'"
+                )
+                req = call.ChangeConfiguration(key=key, value=all_measurands)
+                resp = await self.call(req)
+                _LOGGER.info(
+                    f"'{self.id}' quirk: disable setting OCPP measurands via the quirk if your charger does not need it anymore"
                 )
-                chgr_measurands = all_measurands
 
-            accepted_measurands = []
-            cfg_ok = [
-                ConfigurationStatus.accepted,
-                ConfigurationStatus.reboot_required,
-            ]
+            chgr_measurands = await self.get_configuration(key)
```

Comment:
If the try/except is kept here then the quirk code above and config setting is no longer required. Noting also my comment below to avoid the get configuration being called prior to setting the measurands.

Reply:
@drc38, interesting suggestion. So, together with the other one on using `call.ChangeConfiguration` directly, we would have:

1. try/except protected `self.get_configuration`
2. forcing configuration by `call.ChangeConfiguration` depending on try/except and other settings

Problem is that it doesn't work on the charger side if the charger configuration is already faulty. During `self.get_configuration` the chargers rebooting that actually leads to new boot notification of the charger to OCPP server and we start calling `post_connect` . And the cycle repeats. That case is reflected in the last log of https://github.com/lbbrhzn/ocpp/issues/1275#issuecomment-2313314865

So, unfortunately, we need a quirk for these chargers.

As the quirk is still needed, I would suggest to keep `self.configuration` as well. Namely, at some point, we maybe can get rid of the quirk (ABB updates firmware, users have chargers "repaired", ...). Then keeping the same way of setting configuration as for other parameters would keep the code consistent.

What do you think about it?

Reply:
Ah ok, I hadn't realised the initial get configuration call caused a charger reboot.

Reply:
You could put `chgr_measurands = await self.get_configuration(key)` inside the `if autodetect_measurands:` statement and include the call change configuration in the else statement. It would result in setting the measurands twice during first connection, but would avoid having to remove the quirk code later on.

Reply:
Sure, we can do it that way. So, to avoid quirk, we can also move chrg_measurands to later stage as in:

```python
            if autodetect_measurands:
                accepted_measurands = []
                cfg_ok = [
                    ConfigurationStatus.accepted,
                    ConfigurationStatus.reboot_required,
                ]

                for measurand in all_measurands.split(","):
                    _LOGGER.debug(f"'{self.id}' trying measurand: '{measurand}'")
                    req = call.ChangeConfiguration(key=key, value=measurand)
                    resp = await self.call(req)
                    if resp.status in cfg_ok:
                        _LOGGER.debug(f"'{self.id}' adding measurand: '{measurand}'")
                        accepted_measurands.append(measurand)

                accepted_measurands = ",".join(accepted_measurands)
            else:
                accepted_measurands = all_measurands
                req = call.ChangeConfiguration(key=key, value=all_measurands)
                resp = await self.call(req)
                _LOGGER.debug(
                    f"'{self.id}' measurands set manually to {accepted_measurands}"
                )

            chgr_measurands = await self.get_configuration(key)
```

Such combination should work if the user has selected correct measurands. The only problem is that we would "normalize" quirky behavior. Obviously, it would have to be explained in the code why it is in such manner and maybe that is sufficient. 

With the code above, quirk is not needed and can be removed.

If you think it is better, I would make the changes in the code (config, tests) and docs. But would like to be sure that this is what is expected and prefer not to have later to change it back :)

Reply:
Let's see what @lbbrhzn prefers

Reply:
I prefer teh code with no quirk

Reply:
OK, then I will rewrite that bit and change the docs accordingly.